docs/development/release.md

```diff
@@ -0,0 +1,9 @@
+# Release Process
```

Comment:
Yeah, let's nuke this.  I don't think this process has been used by anyone - pure theory.